query-engine/query-engine-node-api/src/engine.rs

```diff
@@ -138,11 +139,25 @@ impl Inner {
 #[napi]
 impl QueryEngine {
     /// Parse a validated datamodel and configuration to allow connecting later on.
+    /// Note: any new method added to this struct should be added to
+    /// `query_engine_node_api::node_drivers::engine::QueryEngineNodeDrivers` as well.
+    /// Unfortunately the `#[napi]` macro does not support deriving traits.
     #[napi(constructor)]
-    pub fn new(napi_env: Env, options: JsUnknown, callback: JsFunction) -> napi::Result<Self> {
+    pub fn new(
+        napi_env: Env,
+        options: JsUnknown,
+        callback: JsFunction,
+        fn_ctx: Option<JsObject>,
```

Comment:
If no `fn_ctx` is passed (i.e., if the 3rd parameter of the JS QueryEngine constructor is `undefined`), this is simply considered as a `None` value.

Reply:
In other words, the previous TS logic for the QueryEngine can remain unaltered (until we're able to ship the `nodeDrivers` feature in preview).